R/get_sample_cn_segments.R

```diff
@@ -0,0 +1,96 @@
+#' @param multiple_samples Set to TRUE to return cn segments for multiple samples specified in `samples_list` parameter. Default is FALSE.
+#' @param sample_list Optional vector of type character with one sample per row, required if multiple_samples is set to TRUE.
+#' @param from_flatfile This parameter does not do anything for this version of get_manta_sv. See [GAMBLR.results::get_sample_cn_segments] for more info.
+#' @param projection Selected genome projection for returned CN segments. Default is "grch37".
+#' @param this_seq_type Seq type for returned CN segments. One of "genome" (default) or "capture".
+#' @param with_chr_prefix Set to TRUE to add a chr prefix to chromosome names. Default is FALSE.
+#' @param streamlined Return a minimal output rather than full details. Default is FALSE.
+#'
+#' @return A data frame of segments for a specific or multiple sample ID(s).
+#'
+#' @import dplyr readr
+#' 
+#' @export
+#'
+#' @examples
+#' #return sample CNN segmetns for one sample
+#' my_segs = get_sample_cn_segments(this_sample_id = "DOHH-2", projection = "hg38", streamlined = TRUE)
+#'
+#' #define some samples
+#' my_samples = c("SU-DHL-4", "DOHH-2")
+#'
+#' #return CN segments for thhese samples
+#' these_segs = get_sample_cn_segments(multiple_samples = TRUE,
+#'                                     sample_list = my_samples)
+#'
+get_sample_cn_segments = function(this_sample_id,
+                                  multiple_samples = FALSE,
+                                  sample_list,
+                                  from_flatfile = NULL,
+                                  projection = "grch37",
+                                  this_seq_type = "genome",
+                                  with_chr_prefix = FALSE,
+                                  streamlined = FALSE){
+  
+  #check seq type
+  if(this_seq_type != "genome"){
+    stop("Currently, only CN segments available for genome samples (in GAMBLR.data). Please run this function with `this_seq_type` set to genome...")
+  }
+
+  #warn/notify the user what version of this function they are using
+  message("Using the bundled CN segments (.seg) calls in GAMBLR.data...")
+  
+  #get invalid parameters for this function
+  invalid_params = c("from_flatfile")
+  
+  #check if any such parameters are provided
+  for(param in invalid_params){
+    if(!is.null(get(param))){
+      print(paste("Unsupported parameter supplied. This is only available in GAMBLR.results:", param))
+      stop()
+    }
+  }
+  
+  #get valid projections
+  valid_projections = grep("meta", names(GAMBLR.data::sample_data), value = TRUE, invert = TRUE)
+  
+  #return CN segments based on the selected projection
+  if(projection %in% valid_projections){
+    all_segs = GAMBLR.data::sample_data[[projection]]$seg
+  }else{
+    stop(paste("please provide a valid projection. The following are available:",
+               paste(valid_projections,collapse=", ")))
+  }
+  
```

Comment:
Why is it set up this way? More of a question about the original GAMBLR function but the bottom line is, why is the user required to tell the function that they want data from >1 sample instead of the function determining that from the length of the argument? Also, why isn't these_samples_metadata an argument for this and the original function?

Reply:
This has been discussed in Slack. To make a long story short, the change requested for this function is currently pending in an open PR in GAMBLR. I have now reworked this function in this package to mimic the proposed execution, parameter names, etc., from that PR.